backend/benches/benchmark.rs

```diff
@@ -0,0 +1,139 @@
+use chrono::{DateTime, TimeDelta, Utc};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rand::Rng;
+use scheduling_backend::data_model::graph::DiscreteGraph;
+use scheduling_backend::scheduling::scheduler::{GlobalSchedulerAlgorithm, SchedulerAlgorithm};
+use scheduling_backend::scheduling::{
+    scheduler::NaiveSchedulerAlgorithm, task_for_scheduler::TaskForScheduler,
+};
+
+use either::*;
+use protocol::{tasks::TaskId, time::Timespan};
+
+struct TaskFactory {
+    task_id: TaskId,
+}
+impl TaskFactory {
+    fn get_new_task_id(&mut self) -> TaskId {
+        let res = self.task_id;
+        self.task_id += 1;
+        res
+    }
+    pub fn new() -> Self {
+        TaskFactory { task_id: 0.into() }
+    }
+    pub fn make_tasks(
+        &mut self,
+        amount: usize,
+        start: DateTime<Utc>,
+        max_effect: f64,
+        max_duration: TimeDelta,
+    ) -> Vec<TaskForScheduler> {
+        let mut res = Vec::new();
+        let mut rng = rand::thread_rng();
+        for _ in 0..amount {
+            let timespan_start = TimeDelta::seconds(0).num_seconds();
+            let timespan_end = max_duration.num_seconds();
+
+            let start_offset = TimeDelta::seconds(rng.gen_range(timespan_start..timespan_end));
+            let end_offset =
+                TimeDelta::seconds(rng.gen_range(start_offset.num_seconds()..timespan_end))
+                    + TimeDelta::seconds(1);
+
+            let start_time = start + start_offset;
+            let end_time = start + end_offset;
+
+            let total_duration = (end_time - start_time).num_seconds();
+            let duration = rng.gen_range(1..=total_duration);
+
+            let effect = rng.gen_range(1.0..=max_effect);
+
+            res.push(TaskForScheduler::new(
+                self.get_new_task_id(),
+                Timespan {
+                    start: start_time,
+                    end: end_time,
+                },
+                duration.into(),
+                effect,
+            ));
+        }
+        res
+    }
+}
+
+fn make_discrete_graph(
```

Comment:
Please split this into two functions instead of taking an either.
Then we can also delete either from Cargo.toml